opal/corelib/runtime.js

```diff
@@ -1347,6 +1356,8 @@
   //
   Opal.defn = function(obj, jsid, body) {
     obj.$$proto[jsid] = body;
+    // super dispatcher find, etc. might need to lookup by function body
+    obj.$$funcbodies[body] = jsid;
```

Comment:
This rings an alarm for me, `[body]` will convert the function to a string and use that as the key, which means that will use the actual source code of the function to identify it. That, to me, is a problem because:

- two methods in the inheritance chain can have the same implementation this defeating this technique (even if I'm aware it's surely not common)
- I'm not sure how standard is to have `Function.prototype.toString()` converto the function to its source code, what if the implementation is changed?
- looks like a hack
- could mislead a future me reading this code

Even if I recognize it's practical, I'd rather prefer to have a more robust system, that actually checks a prototype chain for the particular method or alternatively assign an `Opal.uid()` to each function and use that instead of the function body.